src/cfddlc_transactions.cpp

```diff
@@ -217,8 +217,12 @@ std::vector<AdaptorPair> DlcManager::CreateCetAdaptorSignatures(
 
   std::vector<AdaptorPair> sigs;
   for (size_t i = 0; i < nb; i++) {
+    std::vector<SchnorrPubkey> r_values;
```

Comment:
Let's check the array size.
(It was previously done by DlcManager::ComputeAdaptorPoint().)
```suggestion
    if (oracle_r_value.size() != msgs[i].size()) {
      throw CfdException(CfdError::kCfdIllegalArgumentError,
                       "Number of r values and messages must match.");
    }
    std::vector<SchnorrPubkey> r_values;
```

Reply:
Actually the point of this PR is to ensure that there can be less messages than r values. I guess you're looking to add a OutOfRange guard to ensure `oracle_r_value.size() >= msgs[i].size()` ?

Reply:
> Actually the point of this PR is to ensure that there can be less messages than r values. I guess you're looking to add a OutOfRange guard to ensure oracle_r_value.size() >= msgs[i].size() ?

That was the case. :pray:
If so, I think it should be as follows:
```suggestion
    std::vector<SchnorrPubkey> r_values;
    size_t max_count = (oracle_r_value.size() > msgs[i].size()) ? msgs[i].size() : oracle_r_value.size();
    for (size_t j = 0; j < max_count; j++) {
```

Reply:
I think it should error if `oracle_r_value.size() < msgs[i].size()` as this would be invalid IMHO

Reply:
Good point. Updated

https://github.com/p2pderivatives/cfd-dlc/pull/34/files#diff-aade2bd579c5db15761851c4fe892c762a8b5086dcb0371406bab662a739d405R220